src/tests/ftest/util/dmg_utils.py

```diff
@@ -206,3 +234,116 @@ def storage_reset(hosts, user=None, hugepages="4096", insecure=True):
+            default.
+
+    Returns:
+        CmdResult: Object that contains exit status, stdout, and other
+            information.
+    """
+    # Create and setup the command
+    dmg = DmgCommand(path)
+    dmg.insecure.value = insecure
+    dmg.hostlist.value = host_port
+    dmg.request.value = "pool"
+    dmg.action.value = "create"
+    dmg.get_action_command()
+    dmg.action_command.group.value = group
+    dmg.action_command.user.value = user
+    dmg.action_command.acl_file.value = acl_file
+    dmg.action_command.scm_size.value = scm_size
+    dmg.action_command.nvme_size.value = nvme_size
+    dmg.action_command.ranks.value = ranks
+    dmg.action_command.nsvc.value = nsvc
+    dmg.action_command.sys.value = sys
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("Pool create command failed: {}".format(details))
+        return None
+
+    return result
+
+
+def pool_destroy(path, host_port, pool_uuid, insecure=True, force=True):
+    """ Execute pool destroy command through dmg tool to servers provided.
+
+    Args:
+        path (str): Path to the directory of dmg binary.
+        host_port (list of str): List of Host:Port where daos_server runs.
+            Use 10001 for the default port number. This number is defined in
+            daos_avocado_test.yaml
+        pool_uuid (str): Pool UUID to destroy.
+        insecure (bool, optional): Insecure mode. Defaults to True.
+        foce (bool, optional): Force removal of DAOS pool. Defaults to True.
+
+    Returns:
+        CmdResult: Object that contains exit status, stdout, and other
+            information.
+    """
+    # Create and setup the command
+    dmg = DmgCommand(path)
+    dmg.insecure.value = insecure
+    dmg.hostlist.value = host_port
+    dmg.request.value = "pool"
+    dmg.action.value = "destroy"
+    dmg.get_action_command()
+    dmg.action_command.pool.value = pool_uuid
+    dmg.action_command.force.value = force
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("Pool destroy command failed: {}".format(details))
+        return None
+
+    return result
+
+
+def get_pool_uuid_from_stdout(stdout_str):
+    """Get Pool UUID from stdout.
+
+    stdout_str is something like:
+    Active connections: [wolf-3:10001]
+    Creating DAOS pool with 100MB SCM and 0B NvMe storage (1.000 ratio)
+    Pool-create command SUCCEEDED: UUID: 9cf5be2d-083d-4f6b-9f3e-38d771ee313f,
+    Service replicas: 0
+
+    This method makes it easy to create a test.
+
+    Args:
+        stdout_str (str): Output of pool create command.
+
+    Returns:
+        str: Pool UUID.
+    """
+    # Find the following with regex. One or more of whitespace after "UUID:"
+    # followed by one of more of number, alphabets, or -. Use parenthesis to
+    # get the returned value.
+    matches = re.findall(r"UUID:\s+([0-9a-f-]+)", stdout_str)
+    return matches[0]
```

Comment:
If for some reason the UUID is not found in `stdout_str` this line will yield a `IndexError: list index out of range` because there is no index 0 of an empty list.  

Does it makes sense for this method to return `None`, and empty string, or raise some sort of exception if the UUID cannot be found?  In any case I think it would be beneficial to handle the `IndexError` exception and provide a more useful message.

Also, do we know for certain that the UUID will always contain lower case alpha characters, or should we include `A-F` in the regex to play it safe?

Reply:
Thanks for the advice. Yes, I think it's better to support the case when UUID isn't found.